task.sql

```diff
@@ -1 +1,38 @@
-# Write your SQL code for the database creation here. Good luck! 
+-- Створення бази даних ShopDB
+CREATE DATABASE IF NOT EXISTS ShopDB;
+USE ShopDB;
+
+-- Створення таблиці Products
+CREATE TABLE IF NOT EXISTS Products (
+    ID INT AUTO_INCREMENT PRIMARY KEY,
+    Name VARCHAR(100) NOT NULL,
+    Description VARCHAR(255),
+    Price DECIMAL(10, 2) NOT NULL,
+    WarehouseAmount INT NOT NULL
+);
+
+-- Створення таблиці Customers
+CREATE TABLE IF NOT EXISTS Customers (
+    ID INT AUTO_INCREMENT PRIMARY KEY,
+    FirstName VARCHAR(50) NOT NULL,
+    LastName VARCHAR(50) NOT NULL,
+    Email VARCHAR(100) NOT NULL,
+    Address VARCHAR(255)
+);
+
+-- Створення таблиці Orders
+CREATE TABLE IF NOT EXISTS Orders (
+    ID INT AUTO_INCREMENT PRIMARY KEY,
+    CustomerID INT,
```

Comment:
The `CustomerID` column in the `Orders` table should allow NULL values since the foreign key constraint uses `ON DELETE SET NULL`. Consider changing the definition to `CustomerID INT NULL`.